examples/complete/variables.tf

```diff
@@ -0,0 +1,356 @@
+    ip_rules                   = []
+    virtual_network_subnet_ids = []
+  }
+}
+
+variable "public_network_access_enabled" {
+  description = " (Optional) Whether public network access is allowed for this Key Vault. Defaults to true."
+  type        = bool
+  default     = true
+}
+
+#########################################
+# Variables related to Resource Names
+#########################################
+variable "resource_names_map" {
+  description = "A map of key to resource_name that will be used by tf-launch-module_library-resource_name to generate resource names"
+  type = map(object({
+    name       = string
+    max_length = optional(number, 60)
+  }))
+
+  default = {}
+}
+variable "environment" {
+  description = "Environment in which the resource should be provisioned like dev, qa, prod etc."
+  type        = string
+}
+
+variable "environment_number" {
+  description = "The environment count for the respective environment. Defaults to 000. Increments in value of 1"
+  type        = string
+  default     = "000"
+}
+
+variable "resource_number" {
+  description = "The resource count for the respective resource. Defaults to 000. Increments in value of 1"
+  type        = string
+  default     = "000"
+}
+
+variable "logical_product_family" {
+  type        = string
+  description = <<EOF
+    (Required) Name of the product family for which the resource is created.
+    Example: org_name, department_name.
+  EOF
+  nullable    = false
+
+  validation {
+    condition     = can(regex("^[_\\-A-Za-z0-9]+$", var.logical_product_family))
+    error_message = "The variable must contain letters, numbers, -, _, and .."
+  }
+}
+
+variable "logical_product_service" {
+  type        = string
+  description = <<EOF
+    (Required) Name of the product service for which the resource is created.
+    For example, backend, frontend, middleware etc.
+  EOF
+  nullable    = false
+
+  validation {
+    condition     = can(regex("^[_\\-A-Za-z0-9]+$", var.logical_product_service))
+    error_message = "The variable must contain letters, numbers, -, _, and .."
+  }
+}
+
+variable "use_azure_region_abbr" {
+  type        = bool
+  description = "Use Azure region abbreviation in the resource name"
+  default     = true
+}
+
+#########################################
+# Variables related to Role assignment
+#########################################
+
+variable "role_assignments" {
+  description = "A map of role assignments to be created"
+  type = map(object({
+    name                 = optional(string)
+    scope                = optional(string)
+    role_definition_name = string
+    principal_id         = optional(string)
+  }))
+  default = {}
+}
+
+###########################################
+# Variables related to private DNS zone
+###########################################
+
+variable "zone_name" {
+  type        = string
+  description = "Name of the private dns zone"
+  default     = "privatelink.vaultcore.azure.net"
```

Comment:
Is is a user configurable value? If not this should be hard coded in the main.tf. Also, check what this value should be for gov cloud. As I have seen the dns zone names differ for both the cloud versions. If they are different may be we leave the default here as is and add a comment in the descipition that this needs to change for gov cloud